### PR TITLE
4.2 Upgrade: Don't assume platformStatus is present in Infra

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -80,18 +81,18 @@ func AddToManager(m manager.Manager) error {
 			}
 		case configv1.GCPPlatformType:
 			log.Info("initializing GCP actuator")
-			if plat.GCP == nil {
+			if infraStatus.PlatformStatus == nil || infraStatus.PlatformStatus.GCP == nil {
```
Comment (abhinavdahiya) [inline pkg/controller/controller.go]: for GCP platformstatus will always be set. since it's a 4.2 platform

Comment (csrwng) [reply pkg/controller/controller.go]: ack... which is why if it's not there we should just exit

Comment (abhinavdahiya) [reply pkg/controller/controller.go]: +1

**Pull request merged**